src/app/extensions/organization-hierarchies/shared/hierarchy-switch/hierarchy-switch.component.ts

```diff
@@ -13,18 +13,44 @@ import { OrganizationGroup } from '../../models/organization-group/organization-
 })
 @GenerateLazyComponent()
 export class HierarchySwitchComponent implements OnInit {
-  groups: Observable<OrganizationGroup[]>;
-  count: Observable<number>;
-  constructor(private facade: OrganizationHierarchiesFacade) {}
+  
+  /**
+   * determines position of dropbox - dropup or dropdown, default is dropdown
+   */
+  @Input() placement: '' | 'up' = '';
+  
+  groups$: Observable<OrganizationGroup[]>;
+  count$: Observable<number>;
+  selectedElement: OrganizationGroup;
 
+  constructor(private facade: OrganizationHierarchiesFacade) {}
+  
   ngOnInit(): void {
-    this.groups = this.facade.groups$;
-    this.count = this.facade.groupsCount$();
+    this.groups$ = this.facade.groups$;
+    this.count$ = this.facade.groupsCount$();
+    this.setSelectedElement();
+  }
+
+  groupSelected(group: OrganizationGroup): void {
+    if (group) {
+      this.facade.selectGroup(group.id);
+    }
+  }
+
+  setSelectedElement(): void {
+    this.facade.getSelectedGroup$.subscribe(el => this.selectedElement = el);
+    if (!this.selectedElement)
+    {
+      this.groups$.subscribe(el => this.selectedElement = el[0]);
+    }
   }
 
-  dispatch(ev: any): void {
-    if (ev?.target?.value) {
-      this.facade.selectGroup(ev.target.value);
+  getName(group: OrganizationGroup): string {
+    let name = group.name;
+    if(name.length>20)
```

Comment:
if customers have another styling they have to change the component file - this is not good
isn't it possible to give the ul a max width and work with the css styling text-overflow: ellipsis?